torchvision/prototype/datasets/utils/_internal.py

```diff
@@ -47,3 +66,39 @@ def read_mat(buffer: io.IOBase, **kwargs: Any) -> Any:
         raise ModuleNotFoundError("Package `scipy` is required to be installed to read .mat files.") from error
 
     return sio.loadmat(buffer, **kwargs)
+
+
+def image_buffer_from_array(array: np.ndarray, *, format: str = "png") -> io.BytesIO:
+    image = PIL.Image.fromarray(array)
+    buffer = io.BytesIO()
+    image.save(buffer, format=format)
+    buffer.seek(0)
+    return buffer
+
+
+class SequenceIterator(IterDataPipe[D]):
+    def __init__(self, datapipe: IterDataPipe[Sequence[D]]):
+        self.datapipe = datapipe
+
+    def __iter__(self) -> Iterator[D]:
+        for sequence in self.datapipe:
+            yield from iter(sequence)
```

Comment:
Should be doable with https://github.com/pytorch/pytorch/blob/2a5116e1599be7d7fa1be9572f47c316716b74c3/torch/utils/data/datapipes/iter/grouping.py#L102. Otherwise we can always send these datapipes upstream.

Reply:
Let's avoid writing new datapipes if they are redundant with upstream blocks already